pandas/tests/frame/test_mutate_columns.py

```diff
@@ -165,6 +165,29 @@ def test_delitem(self):
         del self.frame['A']
         self.assertNotIn('A', self.frame)
 
+    def test_delitem_multiindex(self):
+        midx = MultiIndex.from_product([['A', 'B'], [1, 2]])
+        df = DataFrame(np.random.randn(4, 4), columns=midx)
+        assert len(df.columns) == 4
+        assert ('A', ) in df.columns
+        assert 'A' in df.columns
+
+        result = df['A']
+        assert isinstance(result, DataFrame)
+        del df['A']
+
+        assert len(df.columns) == 2
+
+        # A still in the levels, BUT get a KeyError if trying
+        # to delete
+        assert ('A', ) not in df.columns
+        with pytest.raises(KeyError):
+            del df[('A',)]
+
+        assert 'A' in df.columns
```

Comment:
As in my other comment, shouldn't we regard this as a bug? 
Not needed to fix in this PR, but I would at least add a FIXME comment here so it is clear this test may change